src/Sentry/Internal/StackFrame.cs

```diff
@@ -113,7 +113,12 @@ public RealStackFrame(StackFrame frame)
 
     public int GetILOffset() => _frame.GetILOffset();
 
-    public MethodBase? GetMethod() => _frame.GetMethod();
+    public MethodBase? GetMethod() =>
+#if TRIMMABLE
+        null;
+#else
+        _frame.GetMethod();
+#endif
```

Comment:
Why conditional though? This has worked before (may have returned null but at least it's not a custom code on our end.

Reply:
Hm... OK this does actually cause us problems.

Because this now always returns `null` when `#TRIMMABLE` is set, we no longer get managed stack frames returned when running tests in `SentryAot.slnf`... which is kind of what we expected - in that context (compiling trimmable code designed to be used with applications being published to AOT, we're not going to be able to use methods like this that rely on reflection so our managed stack traces won't work). 

However, we're also not able to get Native stack frames. When running tests in `SentryAot.slnf`, this returns null :
https://github.com/getsentry/sentry-dotnet/blob/e6bedf6b3959264a5822dc8ea5af9dd61bd0b401/src/Sentry/Internal/DebugStackTrace.cs#L237-L242

Which in turn is because here we just delegate to the dotnet extension method:
https://github.com/getsentry/sentry-dotnet/blob/7000964395cddd5107b954131bdc1d74970c3ff4/src/Sentry/Internal/StackFrame.cs#L128

And that [will always return false](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Diagnostics/StackFrameExtensions.cs):
```csharp
    public static partial class StackFrameExtensions
    {
        public static bool HasNativeImage(this StackFrame stackFrame)
        {
            return stackFrame.GetNativeImageBase() != IntPtr.Zero;
        }
        ...
        public static IntPtr GetNativeImageBase(this StackFrame stackFrame)
        {
            return IntPtr.Zero;
        }
    }
```

I think that's also reasonable. Although we've set the `#TRIMMABLE` constant, the xUnit test runner that is running this test won't actually be published to AOT and compiled to native code... so I would expect `HasNativeImage` to return false in that context. 

I think utlimately there are aspects of the fuctionality that we're building which cannot be tested from xUnit. The only way to test this stuff would be to publish an AOT app that ran various tests (most likely not using xUnit, as I don't think it was designed for that. 

I can't think of a way to [get this working](https://github.com/getsentry/sentry-dotnet/blob/2a8813ab6725caac4bc3a76f71af542ffce7f588/test/Sentry.Tests/Internals/MainExceptionProcessorTests.cs#L120) in xUnit, for example. 🤔 
```suggestion
    public MethodBase? GetMethod() =>
#if TRIMMABLE
        null;
#else
        _frame.GetMethod();
#endif
```

Reply:
I've linked this from the other PR for Trimming for net7.0 but let's reiterate: currently,  there's no xUnit runner to support AOT so no way to test this in unit tests. There's a tracking issue: https://github.com/dotnet/runtime/issues/91069

I second that for now we need to do an integration test for AOT stack frames and rely on unit tests on the managed solution.